cmd/run/logger/logger.go

```diff
@@ -0,0 +1,91 @@
+package logger
+
+import (
+	"bufio"
+	"io"
+	"sync"
+)
+
+var Enabled bool
+
+var bufferSize = 4096
+
+type Logger struct {
+	wmu sync.Mutex
+	pr  *io.PipeReader
+	pw  *io.PipeWriter
+
+	rmu     sync.RWMutex
```

Comment:
hmm, on second thoughts, `rmu` and `wmu` feel misleading as a name because they're both protecting writes at different places. how about this:

```go
type Logger struct {
  ingest struct {
    mu sync.Mutex
    pw *io.PipeWriter
  }

  store struct {
    mu  sync.RWMutex
    arr [4096]string
    off int
  }
}
```

(rename `Loop` to `loop` and start it as a part of `New` so that you don't need to store the `*io.PipeReader` as a field?)